dhctl/pkg/operations/mirror/deckhouse_releases.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2024 Flant JSC
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mirror
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/Masterminds/semver/v3"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/layout"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/yaml"
+
+	"github.com/deckhouse/deckhouse/dhctl/pkg/apis/v1alpha1"
+)
+
+func GenerateDeckhouseReleaseManifests(
+	versionsToMirror []semver.Version,
+	pathToManifestYaml string,
+	releaseChannelsImagesLayout layout.Path,
+) error {
+	manifests := &bytes.Buffer{}
+	manifests.Grow(3 * 1024) // 3 KiB
```

Comment:
why 3 kb?

Please add comment here

Reply:
No scientific explanation here, I just felt like 3 kb would fit most of manifests, so I preallocated the buffer ahead of time to avoid reallocs.